### DISCUSSION TITLE
Failed to start session within `extendHttpServer`

### DISCUSSION THREAD

**Discussion by LabLamb** [keystonejs/keystone#9067] [Questions] [answered]
Im trying to implement a socket server using extendHttpServer and I am trying to authenticate users (If I pass the cookie how do I check the cookie?) on socket connect.
on my connection event I am trying to run:
```
        const { authenticateUserWithPassword } = await context.graphql.run<
          any,
          { email: string; password: string }
        >({
          query: `mutation AUTH($email: String!, $password: String!) {
            authenticateUserWithPassword(email: $email, password: $password) {
              __typename
              ... on UserAuthenticationWithPasswordSuccess {
                item {
                  id
                }
              }
            }
          }`,
          variables: {
            email: userEmail,
            password: userPassword,
          },
        });
```

It’s giving me Failed to start session.

How do I make this work? I am aware this is not the right way to check if the user exist. (I dont need the session to be created actually)

**Comment by dcousens** [comment]
@LabLamb can you please show us exactly the error you are getting?

**Reply by LabLamb** [reply]
Yes, sorry about that.

I got:

```
[Object: null prototype] {
  __typename: 'UserAuthenticationWithPasswordFailure',
  message: 'Failed to start session.'
}
```

**Reply by dcousens** [reply]
https://github.com/keystonejs/keystone/blob/29e960d0bf71d734ebf86fbe46f116053c8fc53e/packages/auth/src/gql/getBaseAuthSchema.ts#L112-L115

What is your `sessionStrategy`?
It appears that `validateSecret` is working, but your `context.sessionStrategy.start` isn't returning the expected type.

**Reply by LabLamb** [reply]
```
const session = statelessSessions({
  maxAge: sessionMaxAge,
  secret: sessionSecret,
});
```

```
const { withAuth } = createAuth({
  listKey: 'User',
  identityField: 'email',
  sessionData: 'name createdAt id',
  secretField: 'password',
  initFirstItem: { ... },
  passwordResetLink: { ... },
    tokensValidForMins: 60,
  },
});
```

**Reply by dcousens** [reply, marked as answer]
@LabLamb how are setting up your `context` to do `context.graphql.run`? That `context` needs to have a `req` and `res` attached using `commonContext.withRequest`

**Reply by LabLamb** [reply]
I didn't modify anything on the content, I simply used the context provided from the callback parameter of `extendHttpServer`.

**Reply by dcousens** [reply]
Please see https://keystonejs.com/docs/config/config#extend-express-app and https://github.com/keystonejs/keystone/blob/main/examples/extend-express-app/keystone.ts for an example of what you will need to do.

Without calling `withRequest`, your `context` has no request attached, and thus will fail to start/attach a session. 
I have recently updated an example in https://github.com/keystonejs/keystone/pull/9068, let me know if anything needs additional explanation.

**Reply by LabLamb** [reply]
Thanks @dcousens I'll be sure the check out that example, very helpful.

I'll update here on my next steps.
